src/state-manager/TransactionConsensus.ts

```diff
@@ -302,14 +359,27 @@ class TransactionConsenus {
           }
 
           const readableReq = deserializeGetTxTimestampReq(requestStream)
+          // handle rare race condition where we have seen the txId but not the timestamp
+          if (Context.config.p2p.timestampCacheFix && this.seenTimestampRequests[readableReq.txId] && this.txTimestampCacheByTxId[readableReq.txId] == null) {
+            nestedCountersInstance.countEvent('consensus', 'get_tx_timestamp seen txId but found no timestamp')
+            return respond(tsReceipt, serializeGetTxTimestampResp)
```

Comment:
what the point serializing empty tsReceipt if result is not found? why not have errors like 400, 404. Or binary protocol is not supporting it ?

Reply:
thanks, have created a new task to look this. (fix will be in a new branch)

Reply:
gold-209